### PR TITLE
fix: [PL-39609]: fix for not updating project and org identifiers

### PR DESCRIPTION
**Pull request by bhavya181**
## Describe your changes
We are able to update project and org ids, if we are updating existing ids changing their case.
Added a check in update call to not update identifiers in such cases
Tested locally by trying to update existing project id by changing its case 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49620)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/impl/OrganizationServiceImpl.java [hunk 1]
```diff
@@ -274,6 +274,10 @@ public Organization update(String accountIdentifier, String identifier, Organiza
             String.format("Update operation not supported for Default Organization (identifier: [%s])", identifier),
             WingsException.USER);
       }
+      if (!existingOrganization.getIdentifier().equals(organizationDTO.getIdentifier())) {
+        throw new InvalidRequestException(
+            String.format("Cannot update identifier of organization [%s]", existingOrganization.getIdentifier()));
+      }
```
Comment (nishantsaini20) [inline 120-ng-manager/src/main/java/io/harness/ng/core/impl/OrganizationServiceImpl.java:280]: Please add UTs for covering the case.

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49620)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/impl/OrganizationServiceImpl.java` line 280 at r1](https://reviewable.io/reviews/harness/harness-core/49620#gh-1239292001:-NYbQgSZEVZRLE5QNfyb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/08b79b8fd51fc018edea5e61bcd9aa7aec029a3b/120-ng-manager/src/main/java/io/harness/ng/core/impl/OrganizationServiceImpl.java#L280)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Please add UTs for covering the case.
</blockquote></details>

Done.

File: 120-ng-manager/src/test/java/io/harness/ng/core/impl/OrganizationServiceImplTest.java [hunk 2]
```diff
@@ -181,6 +184,28 @@ public void testUpdateNonExistentOrganization() {
     assertNull(updatedOrganization);
   }
 
+  @Test
+  @Owner(developers = BHAVYA)
+  @Category(UnitTests.class)
+  public void testUpdateOrgIdentifier_changingItsCase() {
```
Comment (gn-harness) [inline 120-ng-manager/src/test/java/io/harness/ng/core/impl/OrganizationServiceImplTest.java:190]: nit: Test case names generally include three parts

`"<functionUnderTest>_<testSetup>_<expectedBehavior>"`

i.e. update_onlyIdentifierChanged_noOp

Comment (gn-harness) [reply 120-ng-manager/src/test/java/io/harness/ng/core/impl/OrganizationServiceImplTest.java:190]: nit:You can consider adding another test

update_identifierAndSomeOtherFieldChanged_onlyOtherFieldIsUpdated

**Review by gn-harness — APPROVED** [review]

File: 120-ng-manager/src/test/java/io/harness/ng/core/impl/OrganizationServiceImplTest.java (hunk 2 above)
Comment (bhavya181) [reply 120-ng-manager/src/test/java/io/harness/ng/core/impl/OrganizationServiceImplTest.java]: done

**Review by ankit-harness — COMMENTED** [review]
> Reviewed 2 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49620)* status: 4 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained.

**Comment by bhavya181** [comment]
trigger uts

**Comment by bhavya181** [comment]
trigger messagecheck

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 2 of 4 files at r3, 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49620)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by nishantsaini20 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49620)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Pull request merged**